packages/clients/textLocator/src/plugins/GeometrySearch/store/actions/setupDrawReaction.ts

```diff
@@ -0,0 +1,35 @@
+import { PolarActionHandler, PolarStore } from '@polar/lib-custom-types'
+import debounce from 'lodash.debounce'
+import { Feature } from 'ol'
+import VectorSource, { VectorSourceEvent } from 'ol/source/Vector'
+import { GeometrySearchGetters, GeometrySearchState } from '../../types'
+
+let debouncedSearchGeometry: PolarActionHandler<
+  GeometrySearchState,
+  GeometrySearchGetters
+>
+
+export function setupDrawReaction(
+  this: PolarStore<GeometrySearchState, GeometrySearchGetters>,
+  { rootGetters, dispatch }
+) {
+  // features added multiple times; avoid overly requesting
+  debouncedSearchGeometry = debounce(
+    (feature) => dispatch('searchGeometry', feature),
+    20
+  ).bind(this)
+
+  // only keep a single feature in the draw tool
+  const drawSource = rootGetters['plugin/draw/drawSource'] as VectorSource
+  let lastFeature: Feature | undefined
+  drawSource.on(['addfeature'], function (event) {
+    const nextFeature = (event as VectorSourceEvent).feature
```

Comment:
```suggestion
  drawSource.on(['addfeature'], function (event: VectorSourceEvent) {
    const nextFeature = event.feature
```

Reply:
![sad_typescript_noises](https://github.com/Dataport/polar/assets/108349707/62af09fb-faaa-4568-b2d9-498ef6b6c16c)

Yes, but in a more real sense, no. I've not checked in depth, but it seems like `.on` only guarantees the general event, and using that as a type will yield a type that has no `.feature`, leading to another problem. I had to use a "trust me it's there I'll check for undefined" somewhere, and decided it's an `as` situation.